scripts/clean-db-dump.sql

```diff
@@ -135,12 +135,61 @@ SET
+                    format('%s-supplier-contract@example.com', substring(id_hash from 0 for 8))
+                END,
+                'contactNameContractNotice', CASE WHEN declaration->'contactNameContractNotice' IS NOT NULL THEN
+                    format('Jo %s. %s. Supplier Esq.', substring(id_hash_ualpha from 8 for 1), substring(id_hash_ualpha from 9 for 1))
+                END,
+                'nameOfOrganisation', CASE WHEN declaration->'nameOfOrganisation' IS NOT NULL THEN
+                    format('%s Ltd.', substring(id_hash_ualpha from 10 for 8))
+                END,
+                'companyRegistrationNumber', CASE WHEN declaration->'companyRegistrationNumber' IS NOT NULL THEN
+                    substring(id_hash from 18 for (substring(id_hash_digits from 19 for 1)::integer + 2))
+                END,
+                'dunsNumber', CASE WHEN declaration->'dunsNumber' IS NOT NULL THEN
+                    substring(id_hash_digits from 23 for 9)
+                END,
+                'registeredVATNumber', CASE WHEN declaration->'registeredVATNumber' IS NOT NULL THEN
+                    substring(id_hash_digits from 14 for 9)
+                END,
+                'tradingNames', CASE WHEN declaration->'tradingNames' IS NOT NULL THEN
+                    format('%s & Co.', substring(id_hash_ualpha from 10 for 8))
+                END,
+                'currentRegisteredCountry', CASE WHEN declaration->'currentRegisteredCountry' IS NOT NULL THEN
+                    initcap(format('%sland', substring(id_hash_ualpha from 26 for 6)))
+                END,
+                'SQ1-1a', CASE WHEN declaration->'SQ1-1a' IS NOT NULL THEN
+                    format('%s Ltd.', substring(id_hash_ualpha from 10 for 8))
+                END,
+                'SQ1-1e', CASE WHEN declaration->'SQ1-1e' IS NOT NULL THEN
+                    substring(id_hash from 18 for (substring(id_hash_digits from 19 for 1)::integer + 2))
+                END,
+                'SQ1-1d-ii', CASE WHEN declaration->'SQ1-1d-ii' IS NOT NULL THEN
+                    initcap(format('%sland', substring(id_hash_ualpha from 26 for 6)))
+                END
+            ))
     END
-WHERE declaration IS NOT NULL
+FROM (
+    -- join against a subquery based on same table to allow us to calculate md5 only once
+    SELECT
+        supplier_id,
+        framework_id,
+        md5(supplier_id::text || '|' || framework_id::text) AS id_hash,
+        upper(translate(md5(supplier_id::text || '|' || framework_id::text), '0123456789', 'IOUYGHJKLM')) AS id_hash_ualpha,
+        translate(md5(supplier_id::text || '|' || framework_id::text), 'abcdef', '987654') AS id_hash_digits
+    FROM supplier_frameworks
+) AS annotation
+WHERE supplier_frameworks.supplier_id = annotation.supplier_id AND supplier_frameworks.framework_id = annotation.framework_id
+  AND declaration IS NOT NULL
   AND json_typeof(declaration) != 'null'
```

Comment:
Could move these in to the sub query for efficiency but doesn't make much odds at all 👍

Reply:
I _think_ the query planner's able to perform that optimization itself